metadata/licenses/private/rules/license.bzl

```diff
@@ -0,0 +1,65 @@
+"""Declares rule `license`."""
+
+load("//:defs.bzl", "PackageAttributeInfo")
+
+def _license_impl(ctx):
+    metadata = ctx.actions.declare_file("{}.package-metadata.attribute.build.bazel.supply-chain.license.json".format(ctx.attr.name))
+
+    ctx.actions.write(
+        output = metadata,
+        content = json.encode({
+            "kind": "build.bazel.supply-chain.license",
+            "label": str(ctx.label),
+        }),
+    )
+
+    return [
+        DefaultInfo(
+            files = depset(
+                direct = [
+                    metadata,
+                ],
+            ),
+        ),
+        PackageAttributeInfo(
+            metadata = metadata,
+            kind = "build.bazel.supply-chain.license",
+            files = depset(
+                direct = [
+                    metadata,
+                ],
+                transitive = [
+                    # TODO(yannic): Define files to propagate here.
+                ],
+            ),
+        ),
+    ]
+
+_license = rule(
+    implementation = _license_impl,
+    attrs = {
+        # TODO
+    },
```

Comment:
Is the idea to eventually have the same attrs as the rules_license implementation of the rules, but to write the values to a json file instead of having the custom LicenseInfo provider?